### PR TITLE
Add JS implementation

### PR DESCRIPTION
**Pull request by siim-m**
- Moved all Python package related files to `python/` subfolder
- JS implementation in `js/`
- Moved test data to `/test_data` (JS tests make use of the same files)
- `js-demo.html` at the root of repo is a basic demo page (added link in main README too). Sources the bbqr package from jsDelivr CDN (with SRI hash checking). Preview in my fork: https://siim-m.github.io/BBQr/js-demo

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -5,3 +5,6 @@ __pycache__
 foo.png
 foo*.svg
 .DS_Store
+
+js/node_modules
+js/dist
```
Comment (doc-hex) [inline .gitignore:10]: I'd prefer these lines in ./js/.gitignore

File: README.md [hunk 2]
```diff
@@ -67,10 +67,10 @@ Created 'example.png' with 8 frames.
...
 # Code Examples
 
-- [spliting QRs](bbqr/split.py)
-- [joining QRs](bbqr/join.py)
-- [binary to internal encoding](bbqr/utils.py)
-- [wrapper CLI](bbqr/cli.py)
+- [spliting QRs](python/bbqr/split.py)
+- [joining QRs](python/bbqr/join.py)
+- [binary to internal encoding](python/bbqr/utils.py)
+- [wrapper CLI](python/bbqr/cli.py)
```
Comment (doc-hex) [inline README.md:73]: add something for the new JS code here

File: .gitignore (hunk 1 above)
Comment (siim-m) [reply .gitignore]: Done

File: README.md (hunk 2 above)
Comment (siim-m) [reply README.md]: Added

**Comment by doc-hex** [comment]
merged and deployed. great work!

**Comment by siim-m** [comment]
Thanks!

**Pull request merged**